auth/src/metadata.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use http::HeaderMap;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use std::env;
+use tokio::time::{self, Duration};
+
+use super::{Error, Result};
+
+// TODO(codyoss): cache a client
+// TODO(codyoss): move client impl elsewhere
+// TODO(codyoss): Instead of vecs things could be slices... don't need ownership
+// TODO(codyoss): funcs could take &str or impl Into<String>?
+
+/// A wrapper around a HTTP client used to talk to a Google Cloud metadata service.
+struct MetadataClient {
```

Comment:
I see no reason for this struct to exist.  How about a factory function instead:
```
fn newMetadataClient() -> reqwest::Client {
```